src/cargo/util/context/mod.rs

```diff
@@ -2321,10 +2321,22 @@ impl ConfigInclude {
                 self.def,
                 abs_path.display(),
             );
-            None
-        } else {
-            Some(abs_path)
+            return Ok(None);
         }
+
+        let reserved_paths = ["config.toml.d", "user.config.toml", "user.config.toml.d"];
```

Comment:
For configuration fragments:

* `conf.d`
* `includes`
* Can't think of others

Reply:
These actually doesn't need to be reserved, and they don't have `.toml` extension so aren't allowed already.